Xamarin.Forms.Platform.Android/Renderers/ShellRenderer.cs

```diff
@@ -356,27 +356,30 @@ void UpdateStatusBarColor(ShellAppearance appearance)
 					color = Color.FromHex("#03A9F4").ToAndroid();
 				}
 
-				Profile.FramePartition("Create SplitDrawable");
-				var split = new SplitDrawable(color, statusBarHeight, navigationBarHeight);
-
-				Profile.FramePartition("SetBackground");
-				decorView.SetBackground(split);
+				if (!(decorView.Background is SplitDrawable splitDrawable) ||
+					splitDrawable.Color != color || splitDrawable.TopSize != statusBarHeight || splitDrawable.BottomSize != navigationBarHeight)
+				{
```

Comment:
All this conditions should be moved to a method to express the intent of what it´s trying to detect